### PR TITLE
Remove support for stub-* files in the manifest

### PR DESCRIPTION
**Pull request by foolip**
The manifest support was added in commit 2650e630d310d720b5eb8f94b39960689cec8160,
ensuring they aren't treated as tests.

Part 2 of https://github.com/web-platform-tests/rfcs/pull/27.

### CODE REVIEW

**Review by jgraham — APPROVED** [review]

**Comment by foolip** [comment]
Thanks for the review, @jgraham. When verifying that an incremental update would work I found that it didn't and pushed some changes to deal with that.

Is that fine, or should we bump the manifest version?

**Comment by jgraham** [comment]
Oh we don't handle a whole test type being dropped. Hmm, seems a shame to leave that code in until we next bump the manifest version, but maybe not too bad. Add a comment indicating it should be removed when the manifest version is higher than whatever the current version is.

@gsnedders may disagree of course.

File: tools/manifest/manifest.py [hunk 1]
```diff
@@ -471,6 +470,12 @@ def from_json(cls, tests_root, obj, types=None):
         self._path_hash = {to_os_path(k): v for k, v in iteritems(obj["paths"])}
 
         for test_type, type_paths in iteritems(obj["items"]):
+            # Drop "stub" items, which are no longer supported but may be
+            # present when doing an incremental manifest update.
+            # See https://github.com/web-platform-tests/rfcs/pull/27 for background.
+            if test_type == "stub":
+                continue
```
Comment (gsnedders) [inline tools/manifest/manifest.py]: Does this result in an incremental update being different to a rebuild? Or do we then add all of them to support when updating?

**Review by gsnedders — COMMENTED** [review]
It seems deliberate that we don't handle test types being removed, though we are pretty inconsistent with validating the JSON conforms to the expected schema.

Can you add a TODO(MANIFESTv7) to the comment so we remember to drop it when we bump the version?

File: tools/manifest/manifest.py (hunk 1 above)
Comment (foolip) [reply tools/manifest/manifest.py]: This just tolerates the and ignores the entry if it exists, but it'll still be dropped when writing a new manifest.

As it turns out, #18745 caused the "stub" entry that this is accepting to be removed, it's not just an empty array. Compare:
https://github.com/web-platform-tests/wpt/releases/download/merge_pr_18690/MANIFEST-ddaab2cf6795cbfa77efdae981b8c494b47e7518.json.bz2
https://github.com/web-platform-tests/wpt/releases/download/merge_pr_18745/MANIFEST-9d89fa83724d2dd6a1cd82975934ff55b8770b42.json.bz2

Confirmed that incremental and full rebuild give the same results.

**Comment by foolip** [comment]
@gsnedders do you want to review again?

**Review by jgraham — APPROVED** [review]

**Comment by foolip** [comment]
@gsnedders merging now, please comment after the fact if you don't like what I did.

File: tools/manifest/manifest.py (hunk 1 above)
Comment (gsnedders) [reply tools/manifest/manifest.py]: > Confirmed that incremental and full rebuild give the same results.

That's all I wanted to be sure of!

**Pull request merged**